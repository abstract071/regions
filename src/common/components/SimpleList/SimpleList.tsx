import React from 'react'
import {
  Link
} from 'react-router-dom'
import Container from '@material-ui/core/Container'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import PublicRoundedIcon from '@material-ui/icons/PublicRounded'


interface ISimpleList {
  values: string[]
}

const SimpleList: React.FC<ISimpleList> = ( {
  values
} ) => {
  return (
    <Container>
      <List component="nav">
        {
          values.map( ( value, idx ) => (
            <ListItem
              key={ `region_${idx}` }
              button
              component={ Link }
              to={ {
                pathname: '/region',
                state: { region: value }
              } }
            >
              <ListItemIcon>
                <PublicRoundedIcon />
              </ListItemIcon>
              <ListItemText primary={ value } />
            </ListItem>
          ) )
        }
      </List>
    </Container>
  )
}

export default SimpleList
