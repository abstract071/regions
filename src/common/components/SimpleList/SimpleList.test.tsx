import React from 'react'
import { shallow } from 'enzyme'

import SimpleList from './SimpleList'


describe( 'SimpleList', () => {
  let shallowedComponent: any = null

  beforeEach( () => {
    shallowedComponent = shallow(
      <SimpleList
        values={ ['Asia', 'Europe'] }
      />
    )
  } )

  it( 'should match snapshot', () => {
    expect( shallowedComponent ).toMatchSnapshot()
  } )
} )
