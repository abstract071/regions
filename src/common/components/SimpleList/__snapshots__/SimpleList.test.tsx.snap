// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleList should match snapshot 1`] = `ShallowWrapper {}`;
